### ISSUE TITLE
`no-unnecessary-arbitrary-value` ignores fractional spacing

### ISSUE THREAD

**Issue by draylegend** [francoismassart/eslint-plugin-tailwindcss#473]
## What's wrong

`no-unnecessary-arbitrary-value` correctly rewrites an arbitrary
spacing value to its preset equivalent when the ratio to the base
spacing unit is a whole integer (e.g. `m-[8px]` → `m-2`, per the rule
docs), but it never proposes anything, no fix and no suggestion, when
the ratio is fractional, even though Tailwind v4's dynamic spacing
utilities support any multiplier via `calc(var(--spacing) * n)`,
including `.25`/`.5`/`.75` steps.

In v4.2.0, `lib/index.mjs`, the spacing-candidate generator:

```js
const re = B / q; // px value ÷ base spacing unit (0.25rem = 4px)
Number.isInteger(re) && N.push(\`${prefix}-${re}${suffix}\`)
```

only pushes a candidate when `re` is a whole integer. Any value landing
on a quarter/half step is silently dropped.

## Repro

With default `--spacing: 0.25rem` (4px) and `cssConfigPath` pointed at
a Tailwind v4 CSS entry:

- `class="p-[16px]"` (16 / 4 = 4, integer) → `eslint --fix` correctly
  rewrites to \`class="p-4"\`.
- `class="p-[14px]"` (14 / 4 = 3.5) → no fix, no suggestion, stays as a
  bare arbitrary-value violation.
- `class="p-[13px]"` (13 / 4 = 3.25) → same as above.
- Writing `class="p-3.25"` directly passes lint clean, confirming it's
  a valid v4 utility the rule just never proposes as a fix.

## Why it matters

Anyone pasting a component from a design/inspiration source with
legacy pixel-based spacing that happens to need a fractional multiplier
gets a hard `no-arbitrary-value` error with no auto-fix and no
suggested replacement — only the manual workaround of computing
`px / spacingUnit` themselves and writing the dynamic class by hand.

## Suggested fix

Drop (or relax) the `Number.isInteger(re)` restriction for the
spacing-candidate path so fractional ratios that Tailwind v4 actually
supports (e.g. quarter-steps) are proposed too, not just integers.

**Comment by draylegend** [comment]
Real-world case this came from: draylegend/letora#13 — same repro (`p-[13px]` / `p-[14px]` not fixed, `p-[16px]` fixed fine), found while wiring up `no-arbitrary-value` + `no-unnecessary-arbitrary-value` as `error` against Angular templates with Tailwind v4.

**Comment by francoismassart** [comment]
It was made on purpose but I think the [future rule `enforces-canonical-classname`](https://github.com/francoismassart/eslint-plugin-tailwindcss/pull/475) should take care of it anyways.

**Comment by nross83** [comment]
I tried `enforces-canonical-classname`, but unfortunately it doesn't seem to address this issue. Perhaps you mean that a future version will address fractional spacing cases? I'm on `v4.4.0`.
